### PR TITLE
Adding the compatibility of graph patterns for the grammar

### PR DESCRIPTION
**Pull request by psvkaushik**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -83,15 +83,25 @@ export interface SelectQuery extends BaseQuery {
...
+    all?: boolean|undefined;
     maxlength?: number| undefined;
     shortest?: boolean | undefined;
     limit?: number | undefined;
     offset?: number | undefined;
     cyclic?: boolean | undefined;
 }
+export interface PathValue {
+    var: Var;
+    value: IriTerm | GraphPattern;
```
Comment (balhoff) [inline index.d.ts]: Value needs to be optional.

File: index.d.ts [hunk 2]
```diff
@@ -83,15 +83,25 @@ export interface SelectQuery extends BaseQuery {
...
+}
+
+export interface PathViaValue {
+    var: Var | null;
+    value: IriTerm | Var | GraphPattern;
```
Comment (balhoff) [inline index.d.ts]:
```suggestion
    value: Var | PropertyPath | Pattern[];
```

File: lib/sparql.jison [hunk 3]
```diff
@@ -769,17 +771,17 @@ PathProperties
     ;
 
 PathProperty
-    : 'START' '=' iri
+    : 'START' Var '=' PathStartValue
```
Comment (balhoff) [inline lib/sparql.jison]: `PathValue` right? Also `= PathValue` should be optional.

File: lib/sparql.jison [hunk 4]
```diff
@@ -769,17 +771,17 @@ PathProperties
...
 
 PathProperty
-    : 'START' '=' iri
+    : 'START' Var '=' PathStartValue 
     {
-      $$ = { property: 'start', value: $3 };
+      $$ = { property: 'start', value: { var: $2, value: $4 } };
     }
-    | 'END' '=' iri
+    | 'END'  Var '=' PathEndValue
```
Comment (balhoff) [inline lib/sparql.jison]: `PathValue` right? Also `= PathValue` should be optional.

File: index.d.ts (hunk 2 above)
Comment (balhoff) [reply index.d.ts]: I changed `GraphPattern` to `Pattern[]`, based on looking at what was used for `WHERE`. Pretty sure `GraphPattern` is for specifying a named graph.

**Review by balhoff — CHANGES_REQUESTED** [review]

**Comment by balhoff** [comment]
Would be good to add tests!

**Comment by psvkaushik** [comment]
- Made the values of Start, End, and Via to be optional. Will include the test cases in the next commit.

File: index.d.ts [hunk 5]
```diff
@@ -83,15 +83,31 @@ export interface SelectQuery extends BaseQuery {
...
     limit?: number | undefined;
     offset?: number | undefined;
     cyclic?: boolean | undefined;
 }
+export interface PathValue {
+    var:{
+        type: Variable,
+        value : string
+    },
+    value: IriTerm | Pattern[];
```
Comment (balhoff) [inline index.d.ts:101]: @psvkaushik shouldn't this be optional? An IRI or a pattern can be supplied but it isn't necessary.

File: index.d.ts [hunk 6]
```diff
@@ -83,15 +83,31 @@ export interface SelectQuery extends BaseQuery {
...
+}
+
+export interface PathViaValue {
+    var: {
+        type: Variable,
+        value : string
+    },
+    value?: IriTerm | Pattern[] | null;
+}
```
Comment (balhoff) [inline index.d.ts:110]: This should just be var or iri or pattern. I don't think you need this structure.

**Review by balhoff — COMMENTED** [review]
@psvkaushik I think there are a couple of issues with this interface. Can you make another PR?

**Pull request merged**